doc/Type/independent-routines.pod6

```diff
@@ -629,16 +629,50 @@ functions.  The only difference between the two functions is that C<sprintf>
 returns a string while the C<printf> function writes to a filehandle. C<sprintf>
 returns a C<Str>, not a literal.
 
-The C<$format> is scanned for C<%> characters. Any C<%> introduces a format
-token. Directives guide the use (if any) of the arguments. When a directive
-other than C<%> is used, it indicates how the next argument passed is to be
-formatted into the string to be created.
+The C<$format> is scanned for C<%> characters. Any C<%> introduces a
+format token. Directives guide the use (if any) of the arguments. When
+a directive other than C<%> is used, it indicates how the next
+argument passed is to be formatted into the string to be
+created. I<Parameter indexes> may also be used in the format
+tokens. They take the form C<N$> and are explained in more detail
+below.
+
+The C<$format> may be defined enclosed in single or double quotes. The
+double-quoted C<$format> string is interpolated before being scanned
+and any embedded string whose interpolated value contains a '%'
+character will cause an exception. N<The invalid sprintf
+formatexception messages have been improved for Rakudo releases after
```

Comment:
>formatexception

No space here? Also, this note seems redundant to me. If the user uses a new compiler, they will see a warn. Otherwise, no. No point to document this implementation detail in docs.

Reply:
Um, why not? If they see the message, they might look it up in Google and they'll be happy to find this.

Reply:
1)If the warning message does not give a clear meaning to the user and they google it up, maybe better to fix the message? We usually have nice ones.
2)Docs describe API user can use in their code: types, routines, syntax, generic techniques etc. Compiler warnings are not API. I mean, technically we have e.g. glossary page and some other places which are not language API, but there is really no need to introduce it here imo.

Reply:
OK, I see your point.

Reply:
I would like to keep the note for now (or file a new doc issue). My intent was to cover this short period before my Rakudo error message change is seen in the next release.

The message shown in the example will have to be changed after the next release. Maybe keep the note for now AND file an issue?

Reply:
> I would like to keep the note for now (or file a new doc issue). My intent was to cover this short period before my Rakudo error message change is seen in the next release.
> 
> The message shown in the example will have to be changed after the next release. Maybe keep the note for now AND file an issue?

OK, your call. I don't think it's superfluous, but I see what @Altai-man says.

Reply:
I am working on the changes. I will commit them in an hour or so after I check them more closely in context.